### PR TITLE
Part2

### PR DESCRIPTION
**Pull request by tjaskula**
Implement Part 2. I'm not sure about the implementation. I was following the output to guide my changes

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -99,14 +100,127 @@ inf
...
+
+    FloatType multiply(float a)
+    {
+        *value *= a;
+        return *this;
+    }
+
+    FloatType divide(float a)
+    {
+        if (a == .0f  || *value == .0f)
```
Comment (matkatmusic) [inline main.cpp]: It doesn't matter if value is 0.  it matters if 'a' is 0.
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -99,14 +100,127 @@ inf
...
+
+    DoubleType multiply(double a)
+    {
+        *value *= a;
+        return *this;
+    }
+
+    DoubleType divide(double a)
+    {
+        if (a == .0 || *value == 0.0)
```
Comment (matkatmusic) [inline main.cpp]: It doesn't matter if value is 0.  it matters if 'a' is 0.
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -99,14 +100,127 @@ inf
...
+
+    IntType multiply(int a)
+    {
+        *value *= a;
+        return *this;
+    }
+
+    IntType divide(int a)
+    {
+        if (a == .0 || value == nullptr)
```
Comment (matkatmusic) [inline main.cpp]: value can never be a nullptr because of how the class is constructed
This nullptr check is unnecessary
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 1 above)
Comment (tjaskula) [reply main.cpp]: Solved

File: main.cpp (hunk 3 above)
Comment (tjaskula) [reply main.cpp]: Solved

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**